eis_toolkit/prediction/cnn_classification_and_probability.py

```diff
@@ -0,0 +1,353 @@
+    dropout or regularization.
+
+    Args:
+         x_train: This is the portion of the data used for training the model.
+         y_train: Partition of the labels used for training: they can be encoded (done with OHE) or a list of integers.
+         x_validation: This is the partition of the dataset used for validation.
+         y_validation: Partition of the labels used for validation in the same form of the y_train.
+         batch_size: How much we want the batch size. This is the number os samples that the CNN takes during each
+                     iteration.
+         epochs: How many iterations we want to run the model.
+         input_shape_for_cnn: Shape of the inputs windows should follow:
+                              - tuple[int, int, int] feed the cnn with windows: format (h, w, c).
+                              - tuple[int, int] feed a network with points: format (value, c).
+         convolutional_kernel_size: Size of the kernel (usually is the last number of the input tuple).
+         pool_size: Size of the pooling layer (How much you want to reduce the dimension of the data).
+         conv_list: List of units used in each convolutional layer.The length of the list is the number of layers.
+         sample_weights: If you want to sample weights. It is used when there is strong imbalance of the data.
+                  neuron_list: This is a list of dense layers used for the last section of the network (fully connected
+                  layers). The length of the list shows the number of layers.
+         neuron_list: This is a list of dense layers used for the last section of the network (fully connected layers).
+                      The length of the list shows the number of layers.
+         dropout_rate: Float number that help avoiding over-fitting. It just randomly drops samples.
+         regularization: Regularization of each  layer. None if you do not want it:
+                         - None if you prefer to avoid regularization.
+                         - L1 for L1 regularization.
+                         - L2 for L2 regularization.
+                         - L1L2 for L1 L2 regularization.
+         data_augmentation: Usable only if the network is fed by windows.
+         optimizer: Loss optimization function, default is Adam.
+         output_units: How many class you have to predicts.
+
+    Return:
+        cnn_model: The compiled model used in this instance.
+        true_value_of_the_labels: The true labels values of the data. (the test section).
+        predicted_values: The predicted labels of the data. (the test section).
+        score: Value of the evaluation of the network.
+
+    Raises:
+        InvalidArgumentTypeException: Argument like train adapter, valid adapter are invalid.
+        InvalidParameterValueException: Argument like convolution list, neuron list are invalid.
+    """
```

Comment:
This docstring is almost correctly formatted! Just modify the lines with dashes to look like other EIS tool docstrings and remove the variable names in the return statement, then it should be good :+1: